common/src/main/java/com/wynntils/services/mapdata/providers/builtin/WaypointsProvider.java

```diff
@@ -32,11 +32,15 @@ public Stream<MapFeature> getFeatures() {
         return PROVIDED_FEATURES.stream();
     }
 
-    public static void resetFeatures() {
+    public void updateWaypoints(List<CustomPoi> customPois) {
+        PROVIDED_FEATURES.forEach(feature -> notifyCallbacks(feature));
         PROVIDED_FEATURES.clear();
+        customPois.forEach(customPoi -> {
+            registerFeature(customPoi);
+        });
     }
```

Comment:
The main problem is, I don't want to update the list of custom pois (or the equivalent JsonMapLocation) by hand. I can't, storage does not allow me. What I want to do is to tell the provider, hey, this is the list of map locations. Do what you want to do with it. This goes the same way for loot chests.

What do you think, how should we do this? In reality, this PR is partially a preparation for for https://github.com/Wynntils/Artemis/pull/2537, where we need to handle these "edge-cases".

Reply:
I'm not sure I understand what the problem should be. You can determine that *some* change has happened, like adding a method call whenever you do storage.touch(). If you then call this updateWaypoints function, it will tell the mapdata service to invalidate all waypoints you have provided before, and then register them all anew.

I think that will solve your problem exactly as you want?

Reply:
Also, I have tried to use the term "waypoint" instead of "custom poi".

Reply:
The problem is that I don't have a "storage is loaded, here is the first/original list of pois". Yes, I can determine and call this when a change has happened. The problem is the initial set.

Reply:
Hm. Maybe we need to actually add some kind of callback for that. It would make sense. The config can be updated from the GUI without the control of the owning class, but that is not the case for storage. However, it can and will be loaded by the manager, without the control of the owning class, and so it seems reasonable that we could get a notification for that.